### PR TITLE
[PHPMD] Support the comma-separated list options as array

### LINKED ISSUE #1076: [PHPMD] Allow array for comma-separated list options
The following options can allow only a comma-separated list value, e.g. `rule: aaa,bbb`:

- `rule`
- `suffixes`
- `exclude`

If supporting also array format (e.g. `rule: [aaa, bbb]`), I belive it is useful.

https://github.com/sider/runners/blob/950af192a21fa6b42256d644d65eece51310b8a0/lib/runners/processor/phpmd.rb#L9-L12

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

This change aims to increase usability when users configure their `sider.yml`. E.g.

```diff
-rule: "cleancode,codesize,controversial,design"
+rule:
+  - cleancode
+  - codesize
+  - controversial
+  - design
```

Also, this change adds a utility method to handle a comma-separated list to the `Processor` class.

> Link related issues, e.g. "Fix #<ISSUE_ID>", "Related to #<ISSUE_ID>", or "None."

Fix #1076

> Check the following items.

- [x] Add a new [changelog](https://github.com/sider/runners/blob/master/CHANGELOG.md) entry if this change is notable.

### CODE REVIEW

File: lib/runners/processor/phpmd.rb [hunk 1]
```diff
@@ -46,50 +48,48 @@ def target_files
...
     end
 
     def strict
       strict = config_linter[:strict] || config_linter.dig(:options, :strict)
-      ["--strict"] if strict
+      strict ? ["--strict"] : []
     end
 
-    def run_analyzer(changes, targets, rule, options)
+    def run_analyzer(changes)
```
Comment (ybiquitous) [inline lib/runners/processor/phpmd.rb:78]: > __[RuboCop]__ Assignment Branch Condition size for run_analyzer is too high. [<9, 59, 6> 59.98/50] ([view](https://sider.review/gh/repos/202486306/pulls/1253#issue-4082547))
>
> <details>
>
> | Rule | Severity |
> | ---- | -------- |
> | `Metrics/AbcSize` | `convention` |
>
> References:
> - <http://c2.com/cgi/wiki?AbcMetric>
> - <https://en.wikipedia.org/wiki/ABC_Software_Metric>
> - <https://docs.rubocop.org/rubocop/cops_metrics.html#metricsabcsize>
>
> You can [close this issue](https://sider.review/gh/repos/202486306/pulls/1253#issue-4082547) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

I think we can allow the level.

File: lib/runners/processor/phpmd.rb [hunk 2]
```diff
@@ -46,50 +48,48 @@ def target_files
...
-      else
-        # NOTE: It assumes that default config xml is located in $HOME
-        (Pathname(Dir.home) / 'sider_config.xml').realpath.to_s
-      end
+      comma_separated_list(rules) || DEFAULT_CONFIG_FILE
     end
 
     def target_dirs
-      Array(config_linter[:target] || './').flat_map { |target| target.split(',') }.join(',')
+      comma_separated_list(config_linter[:target] || DEFAULT_TARGET)
```
Comment (yykamei) [inline lib/runners/processor/phpmd.rb:55]: Should `target_dirs` always return a String value? If so, how about this?

```suggestion
      comma_separated_list(config_linter[:target]) || DEFAULT_TARGET
```

Comment (ybiquitous) [reply lib/runners/processor/phpmd.rb]: Thanks. Your suggestion is right! 👍

**Comment by ybiquitous** [comment]
Thank you for the review. 😄

**Pull request merged**